Client/include/ChronicleMetadataRPCClient.h

```diff
@@ -0,0 +1,110 @@
+//
+// Created by kfeng on 12/12/22.
+//
+
+#ifndef CHRONOLOG_CHRONICLEMETADATARPCCLIENT_H
+#define CHRONOLOG_CHRONICLEMETADATARPCCLIENT_H
+
+#include <iostream>
+#include <unordered_map>
+#include <functional>
+#include <thallium.hpp>
+#include <sys/types.h>
+#include <unistd.h>
+#include "macro.h"
+#include "RPCFactory.h"
+
+class ChronicleMetadataRPCClient {
+public:
+    ChronicleMetadataRPCClient() {
+        LOGD("%s constructor is called", typeid(*this).name());
+        rpc = ChronoLog::Singleton<ChronoLogRPCFactory>::GetInstance()
+                ->GetRPC(CHRONOLOG_CONF->RPC_BASE_SERVER_PORT);
+        func_prefix = "ChronoLog";
+        switch (CHRONOLOG_CONF->RPC_IMPLEMENTATION) {
+            case CHRONOLOG_THALLIUM_SOCKETS:
+            case CHRONOLOG_THALLIUM_TCP:
+            case CHRONOLOG_THALLIUM_ROCE:
+                func_prefix += "Thallium";
+                break;
+        }
+        LOGD("%s constructor finishes, object created@%p in thread PID=%d",
+             typeid(*this).name(), this, getpid());
+    }
+
+    ~ChronicleMetadataRPCClient() = default;
+
+    int CreateChronicle(std::string &name,
+                         const std::unordered_map<std::string, std::string> &attrs,
+                         int &flags) {
+        LOGD("%s is called in PID=%d, with args: name=%s, flags=%d, attrs=",
+             __FUNCTION__, getpid(), name.c_str(), flags);
+        for (auto iter = attrs.begin(); iter != attrs.end(); ++iter) {
+            LOGD("%s=%s", iter->first.c_str(), iter->second.c_str());
+        }
+        return CHRONOLOG_RPC_CALL_WRAPPER("CreateChronicle", 0, int, name, attrs, flags);
```

Comment:
Shouldn't we check the return values of these RPCs for error codes and report them to the client? Throw an exception and terminate the client program if the RPC returns error codes?

Reply:
try-catch is added at the Visor side now. Different error code in the return value will indicate the error. By this macro, the return value of RPC will be forwarded to client apps directly. Users are required for check return values before moving forward.

Reply:
Will you add this to the design document? Also mention places in the code where you added try-catch to terminate/abort the program vs return error code and continue execution.